src/internal/miscutil/log.go

```diff
@@ -0,0 +1,22 @@
+package miscutil
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// TODO: refactor into a common logging utility.
+func LogStep(name string, cb func() error) (retErr error) {
```

Comment:
Not really a fan of this.  I think it would be more idiomatic to return a closure.  Don't need to change it as part of this PR, but I don't think we should port any more logging to this function.